src/modules/grubcfg/main.py

```diff
@@ -190,62 +215,45 @@ def modify_grub_default(partitions, root_mount_point, distributor):
+                existing_params = re.sub(r"^GRUB_CMDLINE_LINUX_DEFAULT\s*=\s*", "", line).strip("\"'").split()
 
                 for existing_param in existing_params:
-                    existing_param_name = existing_param.split("=")[0]
-
-                    # the only ones we ever add
-                    if existing_param_name not in [
-                            "quiet", "resume", "splash"]:
+                    existing_param_name = existing_param.split("=")[0].strip()
+
+                    # Ensure we aren't adding duplicated params
+                    param_exists = False
+                    for param in kernel_params:
+                        if param.split("=")[0].strip() == existing_param_name:
+                            param_exists = True
+                            break
+                    if not param_exists and existing_param_name not in ["quiet", "resume", "splash"]:
                         kernel_params.append(existing_param)
 
-                lines[i] = kernel_cmd
-                have_kernel_cmd = True
-            elif (lines[i].startswith("#GRUB_DISTRIBUTOR")
-                  or lines[i].startswith("GRUB_DISTRIBUTOR")):
-                if libcalamares.job.configuration.get("keep_distributor", False):
-                    lines[i] = distributor_line
-                    have_distributor_line = True
-                else:
-                    # We're not updating because of *keep_distributor*, but if
-                    # this was a comment line, then it's still not been set.
-                    have_distributor_line = have_distributor_line or not lines[i].startswith("#")
-            # If btrfs or f2fs is used, don't save default
-            if no_save_default and lines[i].startswith("GRUB_SAVEDEFAULT="):
-                lines[i] = "#GRUB_SAVEDEFAULT=\"true\""
-    else:
-        lines = []
+            elif line.startswith("GRUB_DISTRIBUTOR") and libcalamares.job.configuration.get("keep_distributor", False):
+                distributor_parts = line.split("=")
+                if len(distributor_parts) > 1:
+                    distributor = distributor_parts[1].strip("'\"")
+
+            # If a filesystem grub can't write to is used, disable save default
+            if no_save_default and line.strip().startswith("GRUB_SAVEDEFAULT"):
+                grub_config_items["GRUB_SAVEDEFAULT"] = "false"
 
+    always_use_defaults = libcalamares.job.configuration.get("always_use_defaults", False)
+
+    # If applicable add the items from defaults to the dict containing the grub config to wirte/modify
+    if always_use_defaults or overwrite or not os.path.exists(default_grub):
         if "defaults" in libcalamares.job.configuration:
-            for key, value in libcalamares.job.configuration[
-                    "defaults"].items():
+            for key, value in libcalamares.job.configuration["defaults"].items():
                 if value.__class__.__name__ == "bool":
```

Comment:
`isinstance(value, bool)` ? (Not your code, I know)

Reply:
fixed